core/controlplane/config/templates/cloud-config-controller

```diff
@@ -1129,22 +1131,99 @@ write_files:
 {{ end }}
 
 {{if .Experimental.NodeDrainer.Enabled}}
+  - path: /srv/kubernetes/manifests/kube-node-drainer-asg-status-updater-de.yaml
+    content: |
+        kind: Deployment
+        apiVersion: extensions/v1beta1
+        metadata:
+          name: kube-node-drainer-asg-status-updater
+          namespace: kube-system
+          labels:
+            k8s-app: kube-node-drainer-asg-status-updater
+        spec:
+          replicas: 1
+          template:
+            metadata:
+              labels:
+                k8s-app: kube-node-drainer-asg-status-updater
+              annotations:
+                scheduler.alpha.kubernetes.io/critical-pod: ''
+            spec:
+              initContainers:
+                - name: hyperkube
+                  image: {{.HyperkubeImage.RepoWithTag}}
+                  command:
+                  - /bin/cp
+                  - -f
+                  - /hyperkube
+                  - /workdir/hyperkube
+                  volumeMounts:
+                  - mountPath: /workdir
+                    name: workdir
+              containers:
+                - name: main
+                  image: {{.AWSCliImage.RepoWithTag}}
+                  env:
+                  - name: NODE_NAME
+                    valueFrom:
+                      fieldRef:
+                        fieldPath: spec.nodeName
+                  command:
+                  - /bin/sh
+                  - -xec
+                  - |
+                    metadata() { wget -O - -q http://169.254.169.254/2016-09-02/"$1"; }
+                    asg()      { aws --region="${REGION}" autoscaling "$@"; }
+
+                    # Hyperkube binary is not statically linked, so we need to use
+                    # the musl interpreter to be able to run it in this image
+                    # See: https://github.com/kubernetes-incubator/kube-aws/pull/674#discussion_r118889687
+                    kubectl() { /lib/ld-musl-x86_64.so.1 /opt/bin/hyperkube kubectl "$@"; }
+
+                    REGION=$(metadata dynamic/instance-identity/document | jq -r .region)
+                    [ -n "${REGION}" ]
+
+                    # Not customizable, for now
+                    POLL_INTERVAL=30
+
+                    # Instance termination detection loop
+                    while sleep ${POLL_INTERVAL}; do
+
+                      # Fetch the list of instances being terminated by their respective ASGs
+                      instances_to_drain=$(asg describe-auto-scaling-groups | jq -r '.AutoScalingGroups[] | select(.Tags[].Key | contains("kube-aws")) | .Instances[] | select(.LifecycleState == "Terminating:Wait") | .InstanceId')
```

Comment:
Thinking again, I could probably also filter by the `kubernetes.io/cluster/{{ .ClusterName }}` tag; this will only return ASGs associated with the current k8s cluster.